### PR TITLE
WIP feat(build): use ngm build over custom solution

### PR DESCRIPTION
**Pull request by valorkin**
hey
here is the simple method to build AoT compatible + bundles version of any package with one simple command ;)
just checkout and do `npm i`
you will see in `dist` folder version ready for publishing

### CODE REVIEW

**Comment by valorkin** [comment]
BTW A lot of files can be removed from project
Did not touched them for now, let us chat first

File: package.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
   "name": "ng2-translate",
   "description": "An implementation of angular translate for Angular 2",
+  "private": true,
```
Comment (ocombe) [inline package.json]: this should probably be removed ;-)

File: config/spec-bundle.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ testing.TestBed.initTestEnvironment(
  * any file that ends with spec.ts and get its path. By passing in true
  * we say do this recursively
  */
-var testContext = require.context('../tests', true, /\.spec\.ts/);
+var testContext = require.context('../src', true, /\.spec\.ts/);
```
Comment (ocombe) [inline config/spec-bundle.js]: Hmm not sure about this ? Why did you need to move the tests to the source folder ?

File: package.json [hunk 3]
```diff
@@ -60,6 +60,8 @@
     "karma-sourcemap-loader": "^0.3.7",
     "karma-webpack": "^1.8.0",
     "loader-utils": "~0.2.16",
+    "npm-run-all": "3.1.1",
```
Comment (ocombe) [inline package.json]: Do I really need npm-run-all ? You don't seem to use it, and it's probably a dependency of npm-submodules if your lib needs it ?

File: package.json (hunk 1 above)
Comment (ocombe) [inline package.json]: Oh wait, this is because you added a new package.json file in the src folder ?

File: config/spec-bundle.js (hunk 2 above)
Comment (valorkin) [inline config/spec-bundle.js]: not necessary, I can revert this. just dir structure I get used to

File: examples/webpack/src/app.component.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import {Component, ChangeDetectionStrategy} from '@angular/core';
-import {TranslateService} from 'ng2-translate/ng2-translate';
+import {TranslateService} from '../../../src/ng2-translate';
```
Comment (valorkin) [inline examples/webpack/src/app.component.ts]: need to revert all this changes

File: package.json (hunk 1 above)
Comment (valorkin) [inline package.json]: yep, this is safety
to not accidentally publish root package.json

File: package.json [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
   "name": "ng2-translate",
   "description": "An implementation of angular translate for Angular 2",
+  "private": true,
   "scripts": {
     "test": "karma start",
     "test-watch": "karma start --singleRun=false --autoWatch=true",
-    "commit": "npm run prepublish && npm test && git-cz",
-    "prepublish": "ngc && npm run build",
-    "build": "webpack && cp bundles/ng2-translate.umd.js bundles/index.js",
+    "commit": "run-s prepublish test git-cz",
```
Comment (valorkin) [inline package.json]: this is usage of `npm-run-all` nice cross platform way to run npm scripts in series or parallel
I had a question about `git-cz` how to use it

File: package.json (hunk 3 above)
Comment (valorkin) [inline package.json]: https://github.com/ocombe/ng2-translate/pull/326/files/b518e2f0d9e0f859f6d149de2b14ce67d0b61304#r88643127

**Comment by valorkin** [comment]
Root tsconfig can be removed

File: package.json [hunk 6]
```diff
@@ -1,12 +1,14 @@
...
+  "private": true,
   "scripts": {
+    "git-cz": "git-cz",
+    "lint": "tslint \"**/*.ts\" -c tslint.json -e \"node_modules/**\" -e \"dist/**\" -e \"temp/**\"",
     "test": "karma start",
     "test-watch": "karma start --singleRun=false --autoWatch=true",
-    "commit": "npm run prepublish && npm test && git-cz",
-    "prepublish": "ngc && npm run build",
-    "build": "webpack && cp bundles/ng2-translate.umd.js bundles/index.js",
+    "commit": "run-s lint prepublish test git-cz",
```
Comment (valorkin) [inline package.json]: here is `npm run all usage`

**Comment by valorkin** [comment]
so... how to use it
1. `npm run build` (runs `ngm build -p src`) - and check dist folder
you have aot compiled version + umd bundles
2. `npm run build -- -w` (or `ngm build -p src -w`) 
3. `ngm publish -p src` will run test, e2e, prepublish build and publish...
